log4j-conversant/src/main/java/org/apache/logging/log4j/conversant/DisruptorProperties.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.conversant;
+
+import com.conversantmedia.util.concurrent.SpinPolicy;
+import org.apache.logging.log4j.core.util.Integers;
+import org.apache.logging.log4j.kit.env.Log4jProperty;
+import org.apache.logging.log4j.status.StatusLogger;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+@NullMarked
+@Log4jProperty(name = "recycler")
```

Comment:
I see two problems with this class:

1. It hijacks property (i.e., `capacity`) of another class: `RecyclerProperties`
1. It conflates namespaces

I suggest improving these aspects as follows:

1. Remove `capacity` from `DisruptorProperties`
2. Rename `DisruptorProperties` to `DisruptorRecyclerProperties`
3. Inject `RecyclerProperties` alongside `DisruptorRecyclerProperties` in the `DisruptorRecyclerFactory` ctor
1. Use `recycler.conversant` as the namespace (ala Spring Boot uses `spring.datasource.hikari` prefix for HikariCP-specific `spring.datasource` properties)

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/2914/commits/20e0582ba8e097c4bab542149e4d193a62c00828

I also change the default recycler capacity from `2C + 1` to `2C` rounded to a power of `2`.